### ISSUE TITLE
file_name rule false positive for Notification.Name

### ISSUE THREAD

**Issue by xuaninbox** [realm/SwiftLint#2325]
file_name rule false positive for Notification.Name

Filename: 

Notification.Name+ABC.swift

```swift
import Foundation

extension Notification.Name {
    ...
}
```
